protocol/proto/encode_decoder.go

```diff
@@ -62,18 +62,12 @@ func (c *encdec) Close() {
 		return
 	}
 	c.isClosed = true
-	if c.rw != nil {
```

Comment:
Who's responsible for managing the lifecycle of the connection?

Reply:
The user of the EncodeDecoder, basically whoever created the connection. I'm basically double closing those connections now (both here and the creator side), the main reason for removing the `Closer` here is because bufio.ReadWriter does not have `Close` method

Reply:
Is it documented anywhere that it's safe to close the same connection multiple times? It seems to be an assumption we make but not sure if that's actually true. Can you check?

Reply:
Double close a connection will return error "use of closed network connection", this change reduces the double closing for this case as well.